pkg/csi_driver/node.go

```diff
@@ -79,16 +80,31 @@ func (s *nodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublish
 	// Mount source
 	source := fmt.Sprintf("%s:/%s", attr[attrIp], attr[attrVolume])
 
+	// FileSystem type
+	fstype := "nfs"
+
 	// Mount options
 	options := []string{}
+
+	// Windows specific values
+	if runtime.GOOS == "windows" {
+		source = fmt.Sprintf("\\\\%s\\%s", attr[attrIp], attr[attrVolume])
+		fstype = "cifs"
+
+		// Login credentials
+
+		options = append(options, s.driver.config.Username)
```

Comment:
we should have these credentials passed in through the NodePublish secret, which gets set in `NodePublishVolumeRequest.secrets`. https://kubernetes-csi.github.io/docs/secrets-and-credentials.html#node-publish-secret

Reply:
Do we need these secrets on unmount too?

Reply:
I'm not sure how I would implement passing through NodePublish secret because [mount_windows.go](https://github.com/kubernetes/kubernetes/blob/5b496fe8f5933de9b53e35780a486c8f7736d02b/pkg/util/mount/mount_windows.go#L94) requires the username and password to be in options.

Secrets aren't needed to unmount.

Reply:
That's fine, we can take the secrets from `NodePublishVolumeRequest` and use them to construct the `options` passed into the mount call.  This way, we can support per-volume credentials.

Reply:
Done.